### PR TITLE
Add example how to switch to jackrabbit

### LINKED ISSUE #315: Add example for jackrabbit PHPCR

### PR DESCRIPTION
**Pull request by alexander-schranz**
| Q | A
| --- | ---
| Fixed tickets | fixes #315 
| License | MIT

#### What's in this PR?

Add example how to switch to jackrabbit

#### Why?

Asked by the community.

### CODE REVIEW

File: bundles/page/jackrabbit.rst [hunk 1]
```diff
@@ -0,0 +1,61 @@
+Jackrabbit
+==========
+
+If you want to have the `versioning`_ feature or want to improve the performance you
+maybe come to the time to change from ``doctrinedbal`` phpcr implementation to ``jackrabbit``.
+
```
Comment (kortwotze) [inline bundles/page/jackrabbit.rst]: Add installation requirements - at least `composer require jackalope/jackalope-jackrabbit`

Comment (alexander-schranz) [reply bundles/page/jackrabbit.rst]: added in the installation part

**Comment by danrot** [comment]
I am not sure if the "Bundles" section is the right place for this documentation, as it also affects other parts... This guide also includes some migration part, which might be quite a good fit for the "Cookbook" section? I actually don't like our cookbook, but until we have found a better structure it might be the place for cross-cutting concerns like this.

And I think I would also link to the new documentation from the [versioning documentation](https://github.com/sulu/sulu-docs/blob/227b376/bundles/page/versioning.rst).

File: bundles/page/jackrabbit.rst [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+.. code-block:: bash
+
+   composer require jackalope/jackalope-jackrabbit
+
+Configuration
+-------------
+
+You need to change the following in ``config/packages/sulu_document_manager.yaml`` file
+its recommended to have the url as an environment variable in your .env file.
```
Comment (danrot) [inline bundles/page/jackrabbit.rst]:
```
Change the ``config/packages/sulu_document_manager.yaml`` file to something similar as below. Mind that it is recommended to pass the URL via an environment variable, which can e.g. be set in your ``.env`` file.

File: bundles/page/jackrabbit.rst [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                backend:
+                    type: jackrabbit
+                    url: "%env(JACKRABBIT_URL)%"
+                workspace: "%env(PHPCR_WORKSPACE)%"
+
+Migration
+---------
+
+If you want to migrate from ``docrinedbal`` to ``jackrabbit`` phpcr implementation before
+switching export the data the following way:
```
Comment (danrot) [inline bundles/page/jackrabbit.rst]:
```
In order to migrate from ``doctrinedbal`` to ``jackrabbit`` you have to export your data before changing the configuration:
```

File: bundles/page/jackrabbit.rst [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+.. code-block:: bash
+
+    bin/adminconsole doctrine:phpcr:workspace:export -p /cmf cmf.xml
+    bin/websiteconsole doctrine:phpcr:workspace:export -p /cmf cmf_live.xml
+    bin/adminconsole doctrine:phpcr:workspace:export -p /jcr:versions jcr.xml
+
+Then switch the configuration to jackrabbit as documented above and then first remove
```
Comment (danrot) [inline bundles/page/jackrabbit.rst]: Then change the configuration as explained in the above Configuration section, and executed these commands to clear any previously existing data (first you should make sure that you really don't need this data anymore).

**Comment by alexander-schranz** [comment]
@danrot updated the PR with your suggestions.

**Comment by danrot** [comment]
All of them unless the one about "Bundles" not being the best place :wink: I think I would go for the "Cookbook" in that case, since this does not only affect the `SuluPageBundle`.

**Comment by alexander-schranz** [comment]
@danrot moved it to cookbook. The sphinx build in travis did fail opened a new pull request #480 to fix this also for older versions of the docs.

File: cookbook/jackrabbit.rst [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+Installation
+------------
+
+Use the following command to install the ``jackrabbit`` adapter:
+
+.. code-block:: bash
+
+   composer require jackalope/jackalope-jackrabbit
+
+In addition to the previous command you also have to make sure that Jackrabbit is running
```
Comment (danrot) [inline cookbook/jackrabbit.rst]: Maybe link to the Jackrabbit documentation about how to install Jackrabbit on your server?

File: cookbook/jackrabbit.rst [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+on your server.
+
+Configuration
+-------------
+
+Change the ``config/packages/sulu_document_manager.yaml`` file to something similar as
+below. Mind that it is recommended to pass the URL via an environment variable, which
```
Comment (danrot) [inline cookbook/jackrabbit.rst]: I would add a link to the Symfony documentation explaining their environment variables.

**Comment by danrot** [comment]
Additionally the `error.log` file should be deleted, and I have already back merged #480, so can you rebase this branch, just to be sure?

**Comment by alexander-schranz** [comment]
@danrot done!

File: bundles/page/versioning.rst [hunk 7]
```diff
@@ -8,7 +8,7 @@ restored, which will result in a new draft.
 
 .. note::
 
-   Mind that this feature is only available for Jackrabbit, because Jackalope does
+   Mind that this feature is only available for :doc:`../../cookbook/jackrabbit`, because Jackalope does
```
Comment (danrot) [inline bundles/page/versioning.rst]: This will write "for Using Jackrabbit", because that's the title of this page....

Comment (alexander-schranz) [reply bundles/page/versioning.rst]: fixed

**Pull request merged**